dotnet/src/SemanticKernel/Planning/ConditionalFlowHelper.cs

```diff
@@ -92,6 +88,31 @@ public async Task<string> IfAsync(string ifFullContent, SKContext context)
             : elseNode?.InnerXml ?? string.Empty;
     }
 
+    /// <summary>
+    /// Get a planner while statement content and output or not its contents depending on the conditional evaluation.
+    /// </summary>
+    /// <param name="whileContent">While content.</param>
+    /// <param name="context"> The context to use </param>
+    /// <returns>None if evaluates to false or the children steps appended of a copy of the while structure</returns>
+    /// <remarks>
+    /// This skill is initially intended to be used only by the Plan Runner.
+    /// </remarks>
+    public async Task<string> WhileAsync(string whileContent, SKContext context)
+    {
+        XmlDocument xmlDoc = new();
+        xmlDoc.LoadXml("<xml>" + whileContent + "</xml>");
+
+        this.EnsureWhileStructure(xmlDoc, out var whileNode);
+
+        var usedVariables = this.GetUsedVariables(whileNode);
+
+        bool conditionEvaluation = await this.EvaluateConditionAsync(whileNode, usedVariables, context).ConfigureAwait(false);
+
+        return conditionEvaluation
+            ? whileNode.InnerXml + whileContent
```

Comment:
What is whileContent? I imagine this will become a Plan object and want to understand the mapping of things.

Reply:
While content is the whileNode.OuterXml (includes the while condition)